src/OrbitGl/CaptureWindowTest.cpp

```diff
@@ -79,4 +81,193 @@ TEST_F(UnitTestCaptureWindow, SlidersRespondToMouseLeave) {
+
+    CreateSchedulerTimers();
+
+    // Prepare the TimeGraph to be fully zoomed out
+    UpdateChildrenPosAndSize();
+    time_graph_->ZoomAll();
+    time_graph_->UpdateLayout();
+    UpdateChildrenPosAndSize();
+
+    // Make sure our expectations about the height of the timeline are correct,
+    // otherwise zooming may not work as expected - if this fails, the overall
+    // viewport height should be changed and the test adjusted accordingly
+    assert(time_graph_->GetTimelineUi()->GetHeight() < 95);
+    assert(time_graph_->GetTimelineUi()->GetPos()[0] == 0);
+  }
+
+ protected:
+  void ExpectInitialState() {
+    UpdateChildrenPosAndSize();
+
+    EXPECT_LT(vertical_slider_->GetLengthRatio(), 1.0);
+    EXPECT_EQ(vertical_slider_->GetPosRatio(), 0.0);
+
+    EXPECT_EQ(slider_->GetLengthRatio(), 1.0);
+    EXPECT_EQ(slider_->GetPosRatio(), 0.0);
+    EXPECT_EQ(time_graph_->GetMaxTimeUs() * 1000, time_graph_->GetCaptureMax());
+    EXPECT_EQ(time_graph_->GetMinTimeUs() * 1000, 0);
+  }
+
+  enum struct PosWithinCapture { kLeft, kRight, kInBetween };
+  void ExpectIsHorizontallyZoomedIn(PosWithinCapture pos) {
+    UpdateChildrenPosAndSize();
+
+    EXPECT_LT(slider_->GetLengthRatio(), 1.0);
+    switch (pos) {
+      case PosWithinCapture::kLeft:
+        EXPECT_EQ(slider_->GetPosRatio(), 0.0);
+        EXPECT_LT(time_graph_->GetMaxTimeUs() * 1000, time_graph_->GetCaptureMax());
+        EXPECT_EQ(time_graph_->GetMinTimeUs() * 1000, 0);
+        break;
+      case PosWithinCapture::kRight:
+        EXPECT_GT(slider_->GetPosRatio(), 0.0);
+        EXPECT_EQ(time_graph_->GetMaxTimeUs() * 1000, time_graph_->GetCaptureMax());
+        EXPECT_GT(time_graph_->GetMinTimeUs() * 1000, 0);
+        break;
+      case PosWithinCapture::kInBetween:
+        EXPECT_GT(slider_->GetPosRatio(), 0.0);
+        EXPECT_LT(time_graph_->GetMaxTimeUs() * 1000, time_graph_->GetCaptureMax());
+        EXPECT_GT(time_graph_->GetMinTimeUs() * 1000, 0);
+        break;
+    }
+  }
+
+ private:
+  CaptureClientAppInterfaceFake capture_client_app_fake_;
+  std::unique_ptr<orbit_client_data::CaptureData> capture_data_;
+
+  void CreateSchedulerTimers() {
+    using orbit_client_protos::TimerInfo;
+    TimerInfo timer;
+    timer.set_start(0);
+    timer.set_end(100);
+    timer.set_thread_id(TrackTestData::kThreadId);
+    timer.set_processor(0);
+    timer.set_depth(0);
+    timer.set_type(TimerInfo::kCoreActivity);
+    time_graph_->ProcessTimer(timer, nullptr);
+  }
+};
+
+TEST_F(NavigationTestCaptureWindow, ZoomTimeWorksAsExpected) {
+  ExpectInitialState();
+
+  const Vec2i kTimeGraphSize = viewport_.WorldToScreen(time_graph_->GetSize());
+
+  // Zoom time: Zoom in and out horizontally, assure we've zoomed in, and zoomed out
+  // again
+  // Test 1: Middle of the capture window
+  int x = kTimeGraphSize[0] / 2;
+  int y = kTimeGraphSize[1] - 5;
+  MouseWheelMoved(x, y, 1, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kInBetween);
```

Comment:
I would rather be more stronger and check that (x,y) is still in the middle of the capture horizontally.

Reply:
I've added this, PTAL. As expected, I can't check the positions exactly, but I'm now checking with a margin of error of 1ns, which should be fine.

Reply:
Thank you very much! :)